src/scripts/main.js

```diff
@@ -1,3 +1,339 @@
 'use strict';
 
-// write your code here
+let board = [];
+const boardSize = 4;
+let score = 0;
+let gameOver = false;
+let gameStarted = false;
+
+const startButton = document.querySelector('.start');
+const restartButton = document.querySelector('.restart');
+const scoreEl = document.querySelector('.game-score');
+
+const startMessage = document.querySelector('.message.message-start');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+
+const gameBoard = document.querySelector('.game-field');
+const rows = gameBoard.querySelectorAll('.field-row');
+
+document.addEventListener('keydown', changeKey);
+
+startButton.addEventListener('click', () => {
+  toggleButton(startButton, restartButton);
+  hideElem(startMessage);
+  startGame();
+}
+);
+
+restartButton.addEventListener('click', () => {
+  restartGame();
+});
+
+function startGame() {
+  resetGame();
+  generateRandomTile();
+  generateRandomTile();
+  upDateBoard();
+
+  gameStarted = true;
+}
+
+function resetGame() {
+  board = [];
+  initializedBoard();
+  score = 0;
+  gameOver = false;
+}
+
+function restartGame() {
+  startGame();
+  hideElem(loseMessage);
+}
+
+function changeKey(evt) {
+  evt.preventDefault();
+
+  if (!gameOver && gameStarted) {
+    switch (evt.key) {
+      case 'ArrowUp':
+        moveUp();
+        break;
+      case 'ArrowDown':
+        moveDown();
+        break;
+      case 'ArrowLeft':
+        moveLeft();
+        break;
+      case 'ArrowRight':
+        moveRight();
+        break;
```

Comment:
Would be great to add default case too
```suggestion
      case 'ArrowUp':
        moveUp();
        break;
        
      case 'ArrowDown':
        moveDown();
        break;
        
      case 'ArrowLeft':
        moveLeft();
        break;
        
      case 'ArrowRight':
        moveRight();
        break;
```